src/compiler/interfaces.ts

```diff
@@ -160,4 +160,5 @@ export interface Var {
 	hoistable?: boolean;
 	subscribable?: boolean;
 	is_reactive_dependency?: boolean;
+	aliased?: boolean;
```

Comment:
What exactly does the `aliased` field mean? I see that it's another thing that can stop variables from being hoisted, but I don't understand what it's checking for.

Reply:
The idea was to have `aliased` mean whether the variable is passed around functions or given other names as to avoid hoisting functions/objects because they could've been modified through these other names.

We'd still have to track all forms of aliasing though.

Reply:
Tracking all forms of aliasing doesn't sound like it would be possible. Is this another situation where we just need to document how the compiler works - like how only certain types of assignments trigger reactivity, rather than embarking on a never-ending journey to catch more cases?

Reply:
Hm, why not? We wouldn't need perfect tracking, just knowing whether the variable escapes a single name -- the _possibility_ it's used elsewhere. The other option would be to disable function hoisting (like currently for objects), since the way it stands functions that shouldn't be hoisted, are.